snapcraft/internal/meta/_manifest.py

```diff
@@ -33,6 +33,14 @@ def annotate_snapcraft(data, parts_dir: str):
             raise errors.InvalidContainerImageInfoError(
                 image_info) from exception
         data['image-info'] = image_info_dict
+    build_info = os.environ.get('LAUNCHPAD_BUILD_INFO')
```

Comment:
I don't think this environment variable name should be Launchpad-specific.  (For example, you might quite reasonably want to do basically the same thing with a Travis build.)

My preference would be to extend the meaning of `SNAPCRAFT_BUILD_INFO` to cover this (since you have to set that environment variable for the manifest to be generated anyway, and it's currently only tested for non-emptiness), but failing that it should at least begin with `SNAPCRAFT_`.